src/pyconverter/xml2py/writer.py

```diff
@@ -295,90 +391,246 @@ def write_source(
+                python_method = command_obj.to_python(custom_functions)
+                fid.write(f"{python_method}\n")
+            fid.close()
+
+            try:
+                exec(command_obj.to_python(custom_functions))
+            except:
+                raise RuntimeError(f"Failed to execute {python_name}.py") from None
+
+    else:
+        import subprocess
+
+        package_structure = {}
+        all_commands = []
+        specific_classes = get_config_data_value(config_path, "specific_classes")
+
+        for command in command_map.values():
+            if command.name in SKIP_XML:
+                continue
+            if command.group is None:
+                continue
+            initial_module_name, initial_class_name = command.group
+            initial_module_name = initial_module_name.replace("/", "")
+            module_name = initial_module_name.replace(" ", "_").lower()
+            module_path = os.path.join(library_path, module_name)
+            if not os.path.isdir(module_path):
+                os.makedirs(module_path)
+                package_structure[module_name] = {}
+            if initial_class_name in specific_classes.keys():
+                initial_class_name = specific_classes[initial_class_name]
+            class_name = initial_class_name.title().replace(" ", "").replace("/", "")
+            file_name = initial_class_name.replace(" ", "_").replace("/", "_").lower()
+            if file_name == "":
+                pass
+
+            file_path = os.path.join(module_path, f"{file_name}.py")
+            if os.path.isfile(file_path) is not True:
+                class_structure = []
+                with open(file_path, "w", encoding="utf-8") as fid:
+                    fid.write(f"class {class_name}:\n")
+                    fid.close()
+            else:
+                class_structure = package_structure[module_name][file_name][1]
+            class_structure.append(command.py_name)
+
+            package_structure[module_name][file_name] = [class_name, class_structure]
+            with open(file_path, "a", encoding="utf-8") as fid:
+                python_method = command.to_python(custom_functions, prefix="    ")
+                # check if there are any imports before the function definition.
+                str_before_def = re.findall(r"[\s\S]*?(?=def)", python_method)[0]
+                output = re.findall(r"((import|from) [^\n]*)", str_before_def)
+                if len(output) == 0:
+                    fid.write(f"{python_method}\n")
+                    fid.close()
+                else:
+                    fid.close()
+                    import_handler(file_path, python_method, output)
 
-    print(f"Commands written to {cmd_path}")
+            all_commands.append(command.name)
+        try:
+            subprocess.run(["python", str(file_path)])
+        except:
+            print(python_method)
+            raise RuntimeError(f"Failed to execute {file_path}") from None
```

Comment:
Prints just before exceptions tends to get very high up in your command line ouput, so it is better to integrate them in the exception message.

```suggestion
        except Exception as e:
            raise RuntimeError(f"Failed to execute '{python_method}' from '{file_path}'.") from e
```